modules/webhook/general.go

```diff
@@ -0,0 +1,185 @@
+		}
+	case api.HookIssueReOpened:
+		text = fmt.Sprintf("[%s] Pull request %s re-opened by %s", repoLink, titleLink, senderLink)
+	case api.HookIssueEdited:
+		text = fmt.Sprintf("[%s] Pull request %s edited by %s", repoLink, titleLink, senderLink)
+	case api.HookIssueAssigned:
+		list := make([]string, len(p.PullRequest.Assignees))
+		for i, user := range p.PullRequest.Assignees {
+			list[i] = linkFormatter(setting.AppURL+user.UserName, user.UserName)
+		}
+		text = fmt.Sprintf("[%s] Pull request %s assigned to %s by %s", repoLink,
+			strings.Join(list, ", "),
+			titleLink, senderLink)
+		color = greenColor
+	case api.HookIssueUnassigned:
+		text = fmt.Sprintf("[%s] Pull request %s unassigned by %s", repoLink, titleLink, senderLink)
+	case api.HookIssueLabelUpdated:
+		text = fmt.Sprintf("[%s] Pull request %s labels updated by %s", repoLink, titleLink, senderLink)
+	case api.HookIssueLabelCleared:
+		text = fmt.Sprintf("[%s] Pull request %s labels cleared by %s", repoLink, titleLink, senderLink)
+	case api.HookIssueSynchronized:
+		text = fmt.Sprintf("[%s] Pull request %s synchronized by %s", repoLink, titleLink, senderLink)
+	case api.HookIssueMilestoned:
+		mileStoneLink := fmt.Sprintf("%s/milestone/%d", p.Repository.HTMLURL, p.PullRequest.Milestone.ID)
+		text = fmt.Sprintf("[%s] Pull request %s milestoned to %s by %s", repoLink,
+			linkFormatter(mileStoneLink, p.PullRequest.Milestone.Title), titleLink, senderLink)
+	case api.HookIssueDemilestoned:
+		text = fmt.Sprintf("[%s] Pull request %s milestone cleared by %s", repoLink, titleLink, senderLink)
+	}
+
+	var attachmentText string
+	if p.Action == api.HookIssueOpened || p.Action == api.HookIssueEdited {
+		attachmentText = p.PullRequest.Body
+	}
+
+	return text, issueTitle, attachmentText, color
+}
+
+func getReleasePayloadInfo(p *api.ReleasePayload, linkFormatter linkFormatter) (text string, color int) {
+	senderLink := linkFormatter(setting.AppURL+p.Sender.UserName, p.Sender.UserName)
+	repoLink := linkFormatter(p.Repository.HTMLURL, p.Repository.FullName)
+	refLink := linkFormatter(p.Repository.HTMLURL+"/src/"+p.Release.TagName, p.Release.TagName)
+
+	switch p.Action {
+	case api.HookReleasePublished:
+		text = fmt.Sprintf("[%s] Release %s created by %s", repoLink, refLink, senderLink)
+		color = greenColor
+	case api.HookReleaseUpdated:
+		text = fmt.Sprintf("[%s] Release %s updated by %s", repoLink, refLink, senderLink)
+		color = yellowColor
+	case api.HookReleaseDeleted:
+		text = fmt.Sprintf("[%s] Release %s deleted by %s", repoLink, refLink, senderLink)
+		color = redColor
+	}
+
+	return text, color
+}
+
+func getIssueCommentPayloadInfo(p *api.IssueCommentPayload, linkFormatter linkFormatter) (string, string, int) {
+	senderLink := linkFormatter(setting.AppURL+p.Sender.UserName, p.Sender.UserName)
+	repoLink := linkFormatter(p.Repository.HTMLURL, p.Repository.FullName)
+	issueTitle := fmt.Sprintf("#%d %s", p.Issue.Index, p.Issue.Title)
+
+	var text, typ, titleLink string
```

Comment:
typ->type

Reply:
It's a keyword of golang, you cannot use that.